app/Model/Build.php

```diff
@@ -2729,4 +2794,376 @@ public function AssignToGroup()
+    {
+        $this->Project = $project;
+    }
+
+    public function GetTestFailedCount()
+    {
+        return $this->TestFailedCount;
+    }
+
+    public function GetBuildType()
+    {
+        return $this->Type;
+    }
+
+    /**
+     * @return array
+     */
+    public function GetCommitAuthors()
+    {
+        // TODO: implement GetCommitAuthors
+        // note: Per Zack: Depending on the type of submission (i.e. test, build error, etc)
+        // this information may not yet be available as it is contained in the update xml
+        // file submission.
+
+        if (!$this->CommitAuthors) {
+            $db = Database::getInstance();
+            $sql = '
+                SELECT
+                    author,
+                    email,
+                    committeremail
+                FROM
+                    updatefile AS uf,
+                    build2update AS b2u
+                WHERE b2u.updateid = uf.updateid
+                AND b2u.buildid = :buildId
+            ';
+            $stmt = $this->PDO->prepare($sql);
+            $stmt->bindParam(':buildId', $this->Id);
+            if ($db->execute($stmt)) {
+                $authors = [];
+                foreach ($stmt->fetchAll(PDO::FETCH_ASSOC) as $row) {
+                    $hasAuthor = !empty($row['email']);
+                    $hasCommitter = !empty($row['committeremail']);
+
+                    if ($hasAuthor) {
+                        $authors[] = $row['email'];
+                    }
+
+                    if ($hasCommitter) {
+                        $authors[] = $row['committeremail'];
+                    }
+
+                    if ($hasAuthor === false && $hasCommitter === false) {
+                        $authors[] = $row['author'];
+                        // IMPORTANT: PLEASE READ
+                        // This is dubious. In testing this value is often expected to be user1kw with no
+                        // associated domain name. Here are the implications of this:
+                        //   * Could be that $row['author'] is a user name with no guarantees that it is also
+                        //     an email address
+                        //   * Could be that $row['author'] is a first and a last name, which clearly does not
+                        //     work here
+                        //   * If it just so happens to be an email address, as it is used in testing, it is
+                        //     unaccompanied by a domain name, further complicating whether or not any messages
+                        //     addressed to it reach its intended destination in practice
+                        //   * Without this value present at least one test will fail and chances are pretty
+                        //     good that others will follow suit.
+                    }
+                }
+                $this->CommitAuthors = array_unique($authors);
+            }
+        }
+        return $this->CommitAuthors;
+    }
```

Comment:
Yeah, I think that makes sense. But it may involve a bit of doing, nothing complex, just making sure that we have i's dotted and t's crossed. Another PR maybe?